src/pages/home/report/ReportActionItemFragment.js

```diff
@@ -86,39 +90,46 @@ const ReportActionItemFragment = (props) => {
                         )
                 );
             }
+            let {html, text} = props.fragment;
```

Comment:
IIRC props destructuring is against our React style guide. It's more verbose but we should just reference `props.fragment.html` and `props.fragment.text` instead. Plus it will simplify the diff of this PR

Reply:
We are not destructuring `props` hehe, but `props.fragment`. Do we not allow any kind of destructuring?

Reply:
I mean, I suppose it's up to interpretation but I think we _do_ allow destructuring, but not when the source of the variable is `props` or `state`. From the [JS style guide](https://github.com/Expensify/App/blob/c67d0d48c050ac0319ad66d38a37596f137844e7/contributingGuides/STYLE.md#destructuring):

> Don't destructure props or state. It makes the source of a given variable unclear. This guideline helps us quickly know which variables are from props, state, or from some other scope.

Reply:
The point of declaring `let html = props.fragment.html` and `let text = props.fragment.text` (or equivalent destructuration) is because a few lines after we may assigned them again:

```js
html = textWithLineBreaks;
text = textWithLineBreaks;
```

I don't think we should do `props.fragment = {...props.fragment, html: textWithLineBreaks, text: textWithLineBreaks};` because we would be modifying the `props` object, and I think this is not advised (I didn't find anything in the  react documentation, but I did find several opinions on that direction).

Would the following change be fine?

```suggestion
            let html = props.fragment.html;
            let text = props.fragment.text;
```

no destructuring, but it is still kind of hiding the source of `html` and `text`... which is the reason why we say we don't want destructuring.

Reply:
I gotcha

> I don't think we should do props.fragment = {...props.fragment, html: textWithLineBreaks, text: textWithLineBreaks}; because we would be modifying the props object, and I think this is not advised (I didn't find anything in the react documentation, but I did find several opinions on that direction).

This makes sense to me – since we're potentially modifying these variables we can split them off from props. And in that case I don't think it makes any difference whether we using the destructuring syntax or not. Going to resolve this.